hbase-server/src/main/java/org/apache/hadoop/hbase/master/region/MasterRegion.java

```diff
@@ -227,7 +227,23 @@ private static HRegion open(Configuration conf, TableDescriptor td, FileSystem f
     if (!walFs.exists(replayEditsDir) && !walFs.mkdirs(replayEditsDir)) {
       throw new IOException("Failed to create replay directory: " + replayEditsDir);
     }
+
+    // Replay any WALs for the Master Region before opening it.
     Path walsDir = new Path(walRootDir, HREGION_LOGDIR_NAME);
+    if (walFs.exists(walsDir)) {
```

Comment:
Please add comment here to explain why this could happen. The old assumption here is that, once the region is created, the directory should be there.